### PR TITLE
Add type hinting in several places (no change to function)

### PR DESCRIPTION
**Pull request by slightlynybbled**
Made a start on adding standard type hints to functions.  Didn't tackle every function since I'm less familiar with some sections of code than others.

### CODE REVIEW

File: src/wireviz/Harness.py [hunk 1]
```diff
@@ -20,16 +20,16 @@ def __init__(self):
...
-    def add_connector(self, name, *args, **kwargs):
+    def add_connector(self, name: str, *args, **kwargs) -> None:
         self.connectors[name] = Connector(name, *args, **kwargs)
 
-    def add_cable(self, name, *args, **kwargs):
+    def add_cable(self, name: str, *args, **kwargs) -> None:
         self.cables[name] = Cable(name, *args, **kwargs)
 
-    def add_bom_item(self, item):
+    def add_bom_item(self, item: str) -> None:
```
Comment (formatc1702) [inline src/wireviz/Harness.py:29]: The item is a `dict`, not a `str`.

File: src/wireviz/Harness.py [hunk 2]
```diff
@@ -20,16 +20,16 @@ def __init__(self):
...
         self.additional_bom_items.append(item)
 
-    def connect(self, from_name, from_pin, via_name, via_pin, to_name, to_pin):
+    def connect(self, from_name: str, from_pin: int, via_name: str, via_pin: (int, str), to_name: str, to_pin: int) -> None:
```
Comment (formatc1702) [inline src/wireviz/Harness.py:32]: `from_pin` and `to_pin` can be `str` too.

File: src/wireviz/Harness.py (hunk 1 above)
Comment (formatc1702) [inline src/wireviz/Harness.py:29]: The item is a `dict`, not a `str`.

File: src/wireviz/Harness.py (hunk 2 above)
Comment (formatc1702) [inline src/wireviz/Harness.py:32]: `from_pin` and `to_pin` can be `str` as well.

File: src/wireviz/Harness.py (hunk 1 above)
Comment (slightlynybbled) [reply src/wireviz/Harness.py]: Thanks.  I haven't used this directly yet.  Fixed in most recent push.

File: src/wireviz/Harness.py (hunk 2 above)
Comment (slightlynybbled) [reply src/wireviz/Harness.py]: Fixed in most recent push

**Pull request merged**